src/leaflike/user/auth.clj

```diff
@@ -0,0 +1,57 @@
+(ns leaflike.user.auth
+  (:require [buddy.hashers :as hashers]
+            [ring.util.response :as res]))
+
+(defonce user-session (atom nil))
+
+(defn- throw-unauthorized
+  [status]
+  (do (when-not (nil? @user-session)
+        (reset! user-session nil))
+      (res/redirect "/login")))
+
+(defn wrap-authorized
+  [handler]
+  (fn [request]
+    (if (nil? @user-session)
```

Comment:
This can be simplified and the nil? check avoided leveraging truthiness https://github.com/nilenso/leaflike/pull/20/files#r159030661